### PR TITLE
test(scheduler): add unit tests for PriorityQueue

### PR DESCRIPTION
**Pull request by sid200727**
> Note: This PR was developed with AI assistance (Claude).

#### What type of PR is this?
/kind test

#### What this PR does / why we need it:
`pkg/scheduler/util/priority_queue.go` has no unit tests. This PR adds unit test coverage for the `PriorityQueue` data structure used in the scheduler.

Tests added:
-> `TestNewPriorityQueue` - verifies initialization and empty state
->  `TestPriorityQueue_PushPop` - verifies priority ordering (highest priority popped first)
->  `TestPriorityQueue_PopEmpty` - verifies nil return on empty queue
->  `TestPriorityQueue_Empty` - verifies Empty() across push/pop lifecycle
->  `TestPriorityQueue_Clone` - verifies deep copy independence from original
->  `TestPriorityQueue_NilLessFn` - verifies no panic with nil LessFn

#### Which issue(s) this PR fixes:
NONE

#### Special notes for your reviewer:
All tests pass locally.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: pkg/scheduler/util/priority_queue_test.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+*/
+package util
+
+import (
+	"testing"
+)
+
+func intLess(a, b interface{}) bool {
+	return a.(int) > b.(int)
+}
```
[the remark this answers is not in the document]
Comment (sid200727) [reply pkg/scheduler/util/priority_queue_test.go]: Fixed, renamed to intHigherPriority to clearly reflect max-heap ordering.

File: pkg/scheduler/util/priority_queue_test.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	if clone.Len() != pq.Len() {
+		t.Fatalf("expected clone length %d, got %d", pq.Len(), clone.Len())
+	}
+
+	// Modifying clone should not affect original
+	clone.Push(100)
+	if pq.Len() == clone.Len() {
+		t.Fatal("expected clone to be independent of original")
+	}
+}
```
[the remark this answers is not in the document]
Comment (sid200727) [reply pkg/scheduler/util/priority_queue_test.go:112]: Strengthened, now pops and verifies all elements from both original and cloned queues after mutation.

File: pkg/scheduler/util/priority_queue_test.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+	if clone.Len() != pq.Len() {
+		t.Fatalf("expected clone length %d, got %d", pq.Len(), clone.Len())
+	}
+
+	// Modifying clone should not affect original
+	clone.Push(100)
+	if pq.Len() == clone.Len() {
+		t.Fatal("expected clone to be independent of original")
+	}
```
[the remark this answers is not in the document]
Comment (sid200727) [reply pkg/scheduler/util/priority_queue_test.go]: Fixed, the test now drains and verifies pop order of both queues independently.

File: pkg/scheduler/util/priority_queue_test.go [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+	pq := NewPriorityQueue(nil)
+	pq.Push(3)
+	pq.Push(1)
+	pq.Push(2)
+
+	if pq.Len() != 3 {
+		t.Fatalf("expected length 3, got %d", pq.Len())
+	}
+	// Just verify it doesn't panic
+	pq.Pop()
```
[the remark this answers is not in the document]
Comment (sid200727) [reply pkg/scheduler/util/priority_queue_test.go]: Fixed, the test now drains the queue completely to exercise all heap restructuring operations.

**Comment by sid200727** [comment]
I reviewed priority_queue.go directly and noticed it had no test coverage. The implementation wraps Go's container/heap with a custom LessFn comparator for max-heap ordering, and the Clone() method does a shallow copy of the backing slice. I wrote these tests to cover the core behaviours and edge cases (nil comparator, clone independence). 
Happy to add more coverage if there are specific scenarios you'd like tested.
